go-controller/pkg/ovn/service.go

```diff
@@ -48,6 +52,16 @@ func (ovn *Controller) syncServices(services []interface{}) {
 			continue
 		}
 
+		// detect if service has endpoints for stale reject ACL check. If there are endpoints, we need to wipe any
+		// old stale ACLs
+		ep, err := ovn.watchFactory.GetEndpoint(service.Namespace, service.Name)
```

Comment:
Does this go directly to the apiserver, or to the informer cache? If it's the apiserver directly, then it'll potentially take a long time to read all of them on startup. If it's the cache, since WatchServices() gets called before WatchEndpoints(), this will always return nothing.

Reply:
After discussion with Tim I'm probably wrong, because we call WaitForCacheSync() which /should/ populate all the caches with the initial set before running any of the Sync handlers.